rockusb/src/libusb.rs

```diff
@@ -256,6 +256,18 @@ impl Transport {
     pub fn reset_device(&mut self, opcode: ResetOpcode) -> Result<()> {
         self.handle_operation(crate::operation::reset_device(opcode))
     }
+
+    /// Get which mode the Rock device is booted.
+    pub fn boot_mode(&self) -> Result<BootMode> {
+        let version = self
+            .handle
+            .device()
+            .device_descriptor()?
+            .device_version()
+            .sub_minor()
+            & 0x1;
+        Ok(version.into())
```

Comment:
Better to inline the conversation from the version bit to mode as it shouldn't just be a generic From